packages/block-editor/src/components/colors/use-colors.js

```diff
@@ -0,0 +1,96 @@
+/**
+ * External dependencies
+ */
+import memoize from 'memize';
+import { startCase } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { Children, cloneElement, useMemo } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import InspectorControls from '../inspector-controls';
+import PanelColorSettings from '../panel-color-settings';
+import { useBlockEditContext } from '../block-edit';
+
+const createComponent = memoize( ( attribute, color ) => ( { children } ) =>
+	// Clone children, setting the style attribute from the color configuration,
+	// if not already set explicitly through props.
+	Children.map( children, ( child ) =>
+		cloneElement( child, {
+			style: { [ attribute ]: color, ...child.props.style },
+		} )
+	)
+);
+
+const createSetColor = memoize( ( setAttributes, name ) => ( newColor ) =>
+	setAttributes( { [ name ]: newColor } )
+);
+
+const InspectorControlsColorPanel = ( { title, colorSettings } ) => (
+	<InspectorControls>
```

Comment:
I guess we don't need to assume the panel is nested inside InspectorControls and we can left the choice for the user.

Reply:
Where else would you render it?

Reply:
In a placeholder for example as part of a flow to initialize a block.

Reply:
I would rather expose that under `InspectorControlsColorPanel.ColorPanel`. I think the inspector usage is the most common.